packages/rocketchat-lib/server/lib/processDirectEmail.js

```diff
@@ -0,0 +1,39 @@
+const reply = require('emailreplyparser').EmailReplyParser;
+
+RocketChat.processDirectEmail = function(email) {
+	email.body = reply.parse_reply(email.body);
+	console.log(email);
+
+	email.headers.to = email.headers.to[0];
+
+	if (email.headers.to.indexOf('<') >= 0 && email.headers.to.indexOf('>') >= 0) {
+		email.headers.to = email.headers.to.split('<')[1].split('>')[0];
+	}
+
+	email.headers.from = email.headers.from[0];
+
+	if (email.headers.from.indexOf('<') >= 0 && email.headers.from.indexOf('>') >= 0) {
+		email.headers.from = email.headers.from.split('<')[1].split('>')[0];
+	}
+
+	email.headers.date = email.headers.date[0];
+
+	email.headers.references = email.headers.references[0].split('@')[0];
+
+	if (email.headers.references.charAt(0) === '<') {
+		email.headers.references = email.headers.references.substr(1);
+	}
+
+	if (email.headers.to.indexOf('+') >= 0) {
+		console.log('Valid Email');
+		email.headers.mid = email.headers.to.split('@')[0].split('+')[1];
```

Comment:
Extract message id from `To` header which is having format `username+messageId@domain`